instructions/section-1/lesson-1-4.md

```diff
@@ -0,0 +1,130 @@
+[< Back to Overview](../../README.md)
+
+# Lesson 1.4: Advanced React Props and State
+
+This assignment will teach you the following:
+
+- Lifting State
+- Controlled Components
+- Props handling
+
+---
+
+## Instructions
+
+### Getting Started:
+
+Merge your pull request from the previous lesson (if you haven't already):
+
+[View tutorial](https://github.com/Code-the-Dream-School/common-instructions/blob/main/common/how-to-merge.md)
+
+Checkout your main branch and pull changes:
+
+    git checkout main
+    git pull
+
+Create a new local branch to work on separate from the `main` branch:
+
+    git checkout -b lesson-1-4
+
+Now, open the project directory in your code editor and continue to the next section.
+
+### Task List:
+
+#### Move Todo List into State
+
+- [ ] Open `/src/App.js`
+- [ ] Create new state variable named `todoList` with setter `setTodoList` and default value of an empty Array
+- [ ] Pass `todoList` state as a prop named `todoList` to the `TodoList` component
+- [ ] Open `/src/TodoList.js`
+- [ ] Add `props` as a parameter to the `TodoList` functional component
+- [ ] Change `todoList` to reference props instead of the hard-coded variable
+- [ ] Delete the hard-coded `todoList` variable
+- [ ] Run your application and view in browser
+    - [ ] Verify that your Todo List is now empty (no list items)
+
+#### Control "Add Todo" Input
+
+- [ ] Open `/src/AddTodoForm.js`
+- [ ] Create new state variable named `todoTitle` with setter `setTodoTitle`
+- [ ] Modify the `<input>` element to be a controlled input
+    - [ ] Add `value` prop equal to `todoTitle` from component props
+    - [ ] Add `onChange` prop equal to `handleTitleChange` function reference (we will declare this function in the next step)
+- [ ] Above the `handleAddTodo` function, declare a new function named `handleTitleChange` that takes `event` as a parameter
+    - [ ] First, retrieve the input value from the `event` object and store in variable named `newTodoTitle`
+    - [ ] Then, call the state setter `setTodoTitle` and pass `newTodoTitle`
+- [ ] In the `handleAddTodo` function, remove the `todoTitle` variable and update `onAddTodo` callback handler to pass our `todoTitle` state variable instead
+- [ ] Run your application and view in browser
+    - [ ] Enter a new todo in "Add Todo" form, submit, and verify that the title appears below
+
+#### Add New Todo to List
```

Comment:
I think my attention span can follow the written instructions up until this point. From here on, my brain kind of tunes out, without seeing what we're doing in real life. I'm such a visual/experiential learner though, so would definitely get more mileage if I were actually following the steps.

Reply:
understandable, I did redo the code as I wrote the instructions so there shouldn't be any typos or missing steps this time